006-admin-server.md

```diff
@@ -0,0 +1,67 @@
+
+The Java GraphQL library allows you to modularise the schema. It has two methods for doing this, the first being a simple merge of multiple schema definition files. Below is an example:
+```
+File adminSchema = loadSchema("admin.graphqls");
+File registrySchema = loadSchema("registry.graphqls");
+
+TypeDefinitionRegistry typeRegistry = new TypeDefinitionRegistry();
+
+typeRegistry.merge(schemaParser.parse(adminSchema));
+typeRegistry.merge(schemaParser.parse(registrySchema));
+``` 
+The second mechanism allows you to extend a type in the schema. For example, if you have the following schema definitions:
+```
+type Topic {
+   name: String
+}
+
+extend type Topic {
+   partitionCount: Integer
+   replicationFactor: Integer
+}
+```
+the result would be the same as defining the following:
+```
+type Topic {
+   name: String
+   partitionCount: Integer
+   replicationFactor: Integer
+}
+```
+The GraphQL parser does not permit type re-definitions so, a name clash in the schema will create an error and the schema will be rejected. The proposal is to use the Java service interface to define a GraphQL module service which loads a schema for that module and a runtime wiring which is the definition that connects the nodes and properties to the implementation. The schema and wiring are then merged to give a single consolidated GraphQL API. This would allow a module to easily create both a REST interface and a GraphQL interface for the module and share a lot of the implementation between the two. 
+
+### Security
+Vert.x networking is based on Netty and works in a very similar manner. When a request is received by the server, it is funneled through a pipe and that pipe has a number of interceptors or handlers that can inspect the contents at that point and ignore or modify it before allowing it to proceed through the pipe. Vert.x also creates a routing context that travels through the pipe with the request and is available for inspection and/or modification by all the handlers. A handler can decided whether the request should be passed on to the next downstream handler or the request is complete and the downstream handlers should be bypassed and they have access to the full request through the routing context. The handlers that are called for a particular request is defined by the Vert.x `Router` which maps patterns to particular handlers. To implement a security handler we add a handler that matches all paths, i.e. the root path (`/`) and ensure that it appears in the router at the top of the list of patterns. When the handler is called, it creates a generic security context and attaches the security context to the routing context making it accessible to downstream handlers.   
```

Comment:
talking about security, how you envisage authn and authz of users for calling REST or GraphQL API?